core/src/main/scala/org/http4s/Fragment.scala

```diff
@@ -0,0 +1,55 @@
+package org.http4s
+
+import cats.{Order, Show}
+import org.http4s.internal.parboiled2.{Parser => PbParser}
+import org.http4s.parser.{Http4sParser, Rfc3986Predicates, Rfc3986Rules}
+import org.http4s.util.{UrlCodingUtils, Writer}
+import scala.math.Ordered
+
+/** The fragment identifier component of a [[org.http4s.Uri]] allows indirect
+  * identification of a secondary resource by reference to a primary resource
+  * and additional identifying information.
+  *
+  * @see https://www.ietf.org/rfc/rfc3986.txt, Section 3.5
+  */
+final class Fragment private (val value: String) extends AnyVal with Ordered[Fragment] {
```

Comment:
I usually end up regretting `AnyVal`, but it seems alright here?

Reply:
I don't know if it's right to flatten out this URI structure into the root package.  I think URI could be its own microlibrary, though the `HttpCodec` type class is useful for testing and macros.

Reply:
I separate `uri` library would be nice perhaps useful in other contexts